exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillOptiq.java

```diff
@@ -713,11 +715,89 @@ private LogicalExpression getDrillFunctionFromOptiqCall(RexCall call) {
         case "date_trunc": {
           return handleDateTruncFunction(args);
         }
+        case "httprequest":
+        case "http_request": {
+          final String QUOTING_IDENTIFIER = context.getPlannerSettings().getOptions().getOption(PlannerSettings.QUOTING_IDENTIFIERS_KEY).string_val;
+
+          // This code resolves aliases in the http_request function.
+          String completeRawPluginName = ((QuotedString) args.get(0)).value;
+          String username = context.getPlannerSettings().getQueryContext().getQueryUserName();
+          AliasRegistryProvider aliasRegistryProvider = context.getPlannerSettings().getQueryContext().getAliasRegistryProvider();
+          AliasRegistry storageAliasRegistry = aliasRegistryProvider.getStorageAliasesRegistry();
+          AliasRegistry tableAliasRegistry = aliasRegistryProvider.getTableAliasesRegistry();
+
+          // Split into plugin and endpoint
+          String[] parts = completeRawPluginName.split("\\.");
```

Comment:
No need to split it manually, please use the `SchemaPath.parseFromString` method.

Reply:
Fixed.